ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/event-handlers.j2

```diff
@@ -9,7 +9,35 @@
 !
 event-handler {{ handler.name }}
 {%         if handler.trigger is arista.avd.defined %}
+{%             if handler.trigger is arista.avd.defined("on-maintenance")
+                  and handler.trigger_on_maintenance.operation is arista.avd.defined
+                  and handler.trigger_on_maintenance.action is arista.avd.defined %}
+{%                 set trigger_on_manitenance_cli =  "trigger " ~ handler.trigger ~ " " ~ handler.trigger_on_maintenance.operation %}
+{%                 if handler.trigger_on_maintenance.bgp is arista.avd.defined %}
+{%                     if handler.trigger_on_maintenance.bgp.peer is arista.avd.defined %}
+{%                         set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ " bgp " ~ handler.trigger_on_maintenance.bgp.peer %}
+{%                     endif %}
+{%                     if handler.trigger_on_maintenance.vrf is arista.avd.defined %}
+{%                         set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ " vrf " ~ handler.trigger_on_maintenance.vrf %}
+{%                     endif %}
```

Comment:
```suggestion
{%                 if handler.trigger_on_maintenance.bgp.peer is arista.avd.defined %}
{%                     set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ " bgp " ~ handler.trigger_on_maintenance.bgp.peer %}
{%                     if handler.trigger_on_maintenance.vrf is arista.avd.defined %}
{%                         set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ " vrf " ~ handler.trigger_on_maintenance.vrf %}
{%                     endif %}
```

Reply:
We can even consider change bgp.peer to bgp_peer to flatten the model a bit.

Reply:
Fixed!